substrate/pkg/controller/substrate/cluster/launchtemplate.go

```diff
@@ -84,10 +85,21 @@ func (l *LaunchTemplate) Create(ctx context.Context, substrate *v1alpha1.Substra
 	} else {
 		logging.FromContext(ctx).Infof("Created launch template %s", aws.StringValue(discovery.Name(substrate)))
 	}
-	if _, err := l.EC2.CreateLaunchTemplateVersionWithContext(ctx, &ec2.CreateLaunchTemplateVersionInput{LaunchTemplateName: discovery.Name(substrate), LaunchTemplateData: launchTemplateData}); err != nil {
+	// Only update the launch template if it's changed
+	hash, err := hashstructure.Hash(launchTemplateData, hashstructure.FormatV2, &hashstructure.HashOptions{SlicesAsSets: true})
+	if err != nil {
+		return reconcile.Result{}, fmt.Errorf("hashing launch template, %w", err)
+	}
+	launchTemplateVersionOutput, err := l.EC2.CreateLaunchTemplateVersionWithContext(ctx, &ec2.CreateLaunchTemplateVersionInput{
```

Comment:
Making sure I understand the flow, so if someone is using the CLI for the first time, LT version will start with 2?
Version 1 will be created when LT gets created and the version will be revised by `CreateLaunchTemplateVersionWithContext` to 2?

Reply:
Yeah it's a bit odd -- agreed.